### PR TITLE
[GNA] Use OV thread_local implementation

### PR DESCRIPTION
**Pull request by tadamowicz**
### Details:
 - Use InferenceEngine::ThreadLocal instead of the local_thread variable to prevent AppVerifier from detecting a memory leak

### Tickets:
 - 124826

### CODE REVIEW

**Review by marcinkusm — APPROVED** [review]

File: src/plugins/intel_gna/src/backend/gna_limitations.cpp [hunk 1]
```diff
@@ -689,7 +689,13 @@ Limitations::Limitations(const DeviceVersion& target) {
 }
 
 void Limitations::init(const DeviceVersion& compile_target) {
-    k_instance = std::shared_ptr<Limitations>(new Limitations(compile_target));
+    auto& localInstance = kInstance.local();
+    localInstance.reset(new Limitations(compile_target));
```
Comment (marcinkacprzak) [inline src/plugins/intel_gna/src/backend/gna_limitations.cpp:693]: Better to use `make_shared`.

**Review by marcinkacprzak — APPROVED** [review]

File: src/plugins/intel_gna/src/backend/gna_limitations.cpp [hunk 2]
```diff
@@ -677,7 +677,7 @@ constexpr uint32_t Limitations::kBytesPerCropElement;
 constexpr uint32_t Limitations::kBytesPerConcatElement;
 constexpr uint32_t Limitations::kMemoryPageSize;
 
-thread_local std::shared_ptr<Limitations> Limitations::k_instance{nullptr};
+InferenceEngine::ThreadLocal<std::shared_ptr<Limitations>> Limitations::kInstance{nullptr};
```
Comment (wangleis) [inline src/plugins/intel_gna/src/backend/gna_limitations.cpp:680]: Please use `ov::threading::ThreadLocal` instead of `InferenceEngine::ThreadLocal`

Comment (tadamowicz) [reply src/plugins/intel_gna/src/backend/gna_limitations.cpp]: ok

File: src/plugins/intel_gna/src/backend/gna_limitations.cpp (hunk 1 above)
Comment (tadamowicz) [reply src/plugins/intel_gna/src/backend/gna_limitations.cpp]: done

**Pull request merged**